spsp-parent/spsp-core/src/main/java/org/interledger/spsp/PaymentPointerResolver.java

```diff
@@ -1,23 +1,56 @@
 package org.interledger.spsp;
 
+import okhttp3.HttpUrl;
+
+import java.util.Objects;
+
+/**
+ * Defines how to resolve a payment pointer.
+ */
 public interface PaymentPointerResolver {
 
   static PaymentPointerResolver defaultResolver() {
     return new DefaultPaymentPointerResolver();
   }
 
   /**
-   * Resolves PaymentPointer to it's endpoint URL per
-   * https://interledger.org/rfcs/0026-payment-pointers/#payment-setup-protocol-receiver-endpoint-resolution
+   * Resolves PaymentPointer to it's endpoint URL.
+   *
+   * @param paymentPointer ilp payment pointer
+   *
+   * @return endpoint url
+   *
+   * @see "https://interledger.org/rfcs/0026-payment-pointers/#payment-setup-protocol-receiver-endpoint-resolution"
+   * @deprecated Prefer the variant of {@link #resolve} that returns an {@link HttpUrl};
+   */
+  @Deprecated
+  default String resolve(final PaymentPointer paymentPointer) {
+    Objects.requireNonNull(paymentPointer);
+    return resolveHttpUrl(paymentPointer).toString();
+  }
+
+  /**
+   * Resolves PaymentPointer to it's endpoint URL.
+   *
    * @param paymentPointer ilp payment pointer
+   *
    * @return endpoint url
+   *
+   * @see "https://interledger.org/rfcs/0026-payment-pointers/#payment-setup-protocol-receiver-endpoint-resolution"
    */
-  String resolve(PaymentPointer paymentPointer);
+  HttpUrl resolveHttpUrl(PaymentPointer paymentPointer);
```

Comment:
Why HttpUrl instead of just URL? Seems like this is a spsp-client implementation detail (that we are using okhttp3 in the DefaultSpspClient) that's leaking into spsp-core. Not a big deal but worth considering if this is ok.

Reply:
After discussing, let's keep HttpUrl. It's also in `SpspClient` and other places.